quay-enterprise/clair.md

```diff
@@ -0,0 +1,274 @@
+# Clair Setup
+
+<img src="img/Clair_horizontal_color.png" class="img-center" alt="Clair Security Scanner"/>
+
+<a href="https://github.com/coreos/clair">Clair</a> is an open source project for the static analysis
+of vulnerabilities in containers images. It can be used as the engine powering [Quay Security Scanner](security-scanning.md)
+to detect vulnerabilities in all images within Quay, as well as notifiying about new vulnerabilities.
+
+## Initial Setup
+
+### Postgres Database
+
+In order to run Clair, a Postgres database is required. It is recommended to have a Postgres database running
+on separate machine(s) with automatic failover.
+
+#### Postgres database for testing
+
+For testing only, a postgres database can be started as follows:
+
+```
+docker run --name postgres -p 5432:5432 -d postgres
+sleep 5
+docker run --rm --link postgres:postgres postgres sh -c 'echo "create database clairtest" | psql -h "$POSTGRES_PORT_5432_TCP_ADDR" -p "$POSTGRES_PORT_5432_TCP_PORT" -U postgres'
+```
+
+The configuration string for this test database is `postgresql://postgres@{DOCKER HOST GOES HERE}:5432/genschema?sslmode=disable`.
+
+### Download the Clair image
+
+Pull the security-enabled Clair image:
+
+```
+docker pull quay.io/coreos/clair-jwt:latest
+```
+
+### Make a configuration directory for Clair
+
+```
+mkdir clair-config
+cd clair-config
+```
+
+## Configure Clair
+
+Clair can run either as a single instance or in high-availability mode. It is recommended to run more than a single instance
+of Clair, ideally in an auto-scaling group with automatic healing.
+
+Create a `config.yaml` file in the config directory with the following contents, filling in all places with `{{ VARIABLE }}`:
```

Comment:
s/filling in all places/replacing appearances of {{VARIABLE}} with the appropriate value/